proto/logic/v1beta2/params.proto

```diff
@@ -89,3 +95,39 @@ message Interpreter {
     (gogoproto.moretags) = "yaml:\"bootstrap\""
   ];
 }
+
+// GasPolicy defines the policy for calculating predicate invocation costs and the resulting gas consumption.
+message GasPolicy {
+  // WeightingFactor is the factor that is applied to the unit cost of each predicate
+  // to yield the gas value.
+  // If not provided or set to 0, the default value of 10000 is used.
```

Comment:
I didn't saw the handling of the `10000` default value.

Besides, isn't it too much? Maybe I don't understand because from I see it would multiply the current consumption by `10000`?

Reply:
Correct. I don't handle the default value correctly. I'll fix that.

The idea about the weighting factor is to multiply the unit cost of each predicate executed to get the gas: `gas consumed = predicate-unit-cost * weighting-factor`. Hence, if the predicate as a unit cost of 1 (by default), then the gas consumed would be 10000.

This allows for adjusting the costs of predicates relative to each other using an arbitrary unit, and then using a coefficient to convert them to gas. This, I believe, simplifies the implementation of an execution cost policy.

But, the question arises about the default value itself... If you think 10k is too much - and I'd tend to agree with you - what value should we consider? Should we leave 1 ?

Reply:
I think this is too much yes because currently running the most simple `Ask` query consumes 2180 gas which weighted by 10000 would exceed the actual maximum gas per block on the nemeton testnet.

If we leave the value to 1 we'll have the exact same gas consumption as the current implementation, which is not irrelevant from my point of view.

Reply:
Makes sense. :+1:

Reply:
It might not be nice to multiply by a decimal number allowing more accuracy on the weighted factor ? Because the minimum factor after 1 is two (obvious, but one will do nothing) and it's a big difference on the gas consumption result.

Reply:
@bdeneux I see the point. However, following @amimart suggestion, I redesigned the parameters to consider 1 unit as a value for both the default predicate costs and the default weighting factor. So, the weighting factor may not necessarily need to be a decimal for now. However, if greater control is needed, we can consider representing the weighting factor as a rational number (numerator/denominator). Nevertheless, I would prefer to address this in a separate PR, after having had some more time to reflect on these aspects, if that's okay with you.